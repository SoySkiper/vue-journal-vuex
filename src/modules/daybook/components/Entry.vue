<template>
    <div class="entry-conteiner mb-3 pointer p-2" @click="$router.push({name: 'entry', params: {id: 10}})">
        <!-- Titulo -->
        <div class="netry-title d-flex">
            <span class="text-success fs-5 fw-bold">24</span>
            <span class="mx-1 fs-5">Noviembre</span>
            <span class="mx-2 fw-light">2022, jueves</span>
        </div>

        <div class="entry-description">
            Ut exercitation mollit nulla consequat occaecat excepteur aliquip aliqua nostrud duis eiusmod tempor nostrud sit. Reprehenderit excepteur officia aute reprehenderit amet nisi deserunt amet. Ullamco aliquip adipisicing nisi sit incididunt eiusmod et sunt do officia sit quis enim.
        </div>

    </div>
</template>

<script>

</script>

<style lang="scss" scoped>
.entry-conteiner{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover{
        background-color: lighten($color:grey, $amount: 45);
        transition: 0.2s all ease-in;

    }

    .entry-description{
        font-size: 12px;
    }
}
</style>