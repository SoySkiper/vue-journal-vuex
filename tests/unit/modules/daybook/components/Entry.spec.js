import { shallowMount } from "@vue/test-utils"
import Entry from "@/modules/daybook/components/Entry.vue"
import { journalState } from "../../../mock-data/test-journal-state"

describe('Pruebas en Entry component', () => {
    //mockRouter
    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount( Entry, {
        props: { 
            entry: journalState.entries[0]
        },
        global: {
            mocks: { 
                $router: mockRouter
            }
        }
    })

    test('debe de hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot();
    });

    test('debe de redireccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-conteiner')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        });
    });

    test('pruebas en las propiedades computadas', () => {
        expect( wrapper.vm.day ).toBe(17)
        expect( wrapper.vm.month ).toBe('Enero')
        expect( wrapper.vm.yearDay ).toBe('17, Ener')
    });
})

