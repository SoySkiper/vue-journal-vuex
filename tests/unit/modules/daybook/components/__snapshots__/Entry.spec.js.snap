// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry component debe de hacer match con el snapshot 1`] = `
<div class="entry-conteiner mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="netry-title d-flex"><span class="text-success fs-5 fw-bold">17</span><span class="mx-1 fs-5">Enero</span><span class="mx-2 fw-light">2023, Martes</span></div>
  <div class="entry-description">Hola mundo desde mock data</div>
</div>
`;
